### PR TITLE
Add golangci-lint.yml

### LINKED ISSUE #16: Add coding standards to CI via golangci-lint
https://golangci-lint.run/

This would create linting standards that all PRs must pass before merging.

Thoughts? PR welcome. I'm also glad to put one in.

### PR DESCRIPTION
**Pull request by jspeed-meyers**
Fix #16

### CODE REVIEW

**Comment by jspeed-meyers** [comment]
Help welcome. Currently stuck. Error below.

```
run golangci-lint
  Running [/home/runner/golangci-lint-1.53.3-linux-amd64/golangci-lint run --out-format=github-actions] in [] ...
  level=warning msg="Failed to discover go env: failed to run 'go env': exit status 2"
  level=error msg="Running error: context loading failed: failed to load packages: failed to load with go/packages: err: exit status 1: stderr: template: main:1: function \"context\" not defined\n"
  
  Error: golangci-lint exit with code 3
```

**Comment by jspeed-meyers** [comment]
Help welcome :)

**Comment by puerco** [comment]
I'm happy to take over this one :+1:

**Comment by jspeed-meyers** [comment]
@puerco, thank you!. I'll re-assign this to you!

**Comment by jspeed-meyers** [comment]
@puerco, let me know if and when you want a review!

**Comment by jspeed-meyers** [comment]
Thank you, @wey-chiang!

**Comment by puerco** [comment]
O wow, I had not noticed this. Thanks for handling it john speed, we can now enable golangci lint

File: cmd/cli/root.go [hunk 1]
```diff
@@ -58,8 +58,8 @@ var RootCmd = &cobra.Command{
 		}
 		logrus.WithField("config", string(v)).Debugf("Protobom config")
 
-		internal.Translate(path, &Cfg)
-		return nil
+		err = internal.Translate(path, &Cfg)
+		return err
```
Comment (puerco) [inline cmd/cli/root.go:62]: nit for later: We should wrap this error or return it directly

File: pkg/sbom/document.go [hunk 2]
```diff
@@ -20,5 +20,6 @@ func NewDocument() *Document {
 
 // GetRootNodes returns the top level nodes of the document
 func (d *Document) GetRootNodes() []*Node {
-	return d.GetRootNodes()
+	// TODO: Implement me
+	return d.GetRootNodes() //nolint:staticcheck
```
Comment (puerco) [inline pkg/sbom/document.go:24]: I think is a bug, I'll fix it in a follow up

File: pkg/writer/serializer_cdx.go [hunk 3]
```diff
@@ -123,7 +123,9 @@ func (s *SerializerCDX) root(ctx context.Context, bom *sbom.Document) (*cdx.Comp
 
 			// TODO(degradation): Here we would document other root level elements
 			// are not added to to document
-			break
+			if true { // temp workaround in favor of adding a lint tag
+				break
+			}
```
Comment (puerco) [inline pkg/writer/serializer_cdx.go:128]: This one is just a stump for an upcoming change that actually handles this.

**Review by puerco — APPROVED** [review]
Cool, there is still some more work to clean the nits but I think the benefits of enabling the linter now are enough that we should merge it as it is. Thanks for handling these JS! Also, there is a bug below (not from you) and this will hide it, so I'll send a PR fixing it ASAP.

**Pull request merged**